libevmasm/Instruction.cpp

```diff
@@ -325,13 +326,20 @@ void dev::solidity::eachInstruction(
 		size_t additional = 0;
 		if (isValidInstruction(instr))
 			additional = instructionInfo(instr).additional;
+
 		u256 data;
-		for (size_t i = 0; i < additional; ++i)
+
+		// fill the data with the additional data bytes from the instruction stream
+		while (additional > 0 && next(it) < _mem.end())
```

Comment:
@ekpyron agreed RE tests, however, I can very easily tell whether or not I am advancing *beyond* the `end()`, because then the `operator++()` invocation raises an exception (on MSVC, debug build).

Now. Suppose I am one off too early, hence, my `<` needs to be changed to a `<=`, that implies that my `it` at one point (the last iteration) will be equal to `_mem.end()` and that leads to the fact, that the next line `*++it` will advance beyond `end` and hence dereference to a value that is undefined. But maybe I am overseeing something.

Reply:
You're checking ``next(it) < _mem.end()`` not ``it < _mem.end()``! You *want* to get ``*it`` for ``_mem.end() - 1`` and if you access that value with ``*++it``, then afterwards ``it`` *will* be equal to ``_mem.end()``. The problem is that the post-condition of the outer ``for`` will still be executed after the outer ``for`` body, if you don't explicitly break. That will happen *before* the ``_it < _mem.end()`` check in the *outer* ``for``. That's why MSVC will complain.

Reply:
Ah, sorry - all the time I had the behavior of ``*it++`` not ``*++it`` in mind for some reason - you're perfectly right, this should be fine now - sorry for the noise :).